### PR TITLE
Provide "Marshaling data" main article with explanation of how to call native code in C#

### LINKED ISSUE #10308: Lacks example of calling from C# !
@vasil-1603 commented on [Sat Feb 02 2019](https://github.com/dotnet/docs.ru-ru/issues/169)

Очень не хватает примера возможных вызовов из C# !

---
#### Сведения о документе

⚠ *Не вносите правки в этот раздел. Это необходимо для связывания страницы сайта docs.microsoft.com с вопросом на GitHub.*

* ID: 678eaef3-eec4-dc52-1234-37e220f84682
* Version Independent ID: 38b09163-65d8-4d6c-1394-401e080ed6f0
* Content: [Маршалинг данных при вызове неуправляемого кода](https://docs.microsoft.com/ru-ru/dotnet/framework/interop/marshaling-data-with-platform-invoke)
* Content Source: [docs/framework/interop/marshaling-data-with-platform-invoke.md](https://github.com/dotnet/docs.ru-ru/blob/live/docs/framework/interop/marshaling-data-with-platform-invoke.md)
* Product: **dotnet-framework**
* GitHub Login: @rpetrusha
* Microsoft Alias: **ronpet**

### PR DESCRIPTION
**Pull request by BartoszKlonowski**
This pull request fixes #10308

It adds the brief explanation of how to call the native, unmanaged library code in C#.

---

Full explanation of how to utilize the library created in the native language is placed in other documents of the marshaling documentation like [Platform Invoke (P/Invoke)](https://docs.microsoft.com/en-us/dotnet/standard/native-interop/pinvoke) and [Marshaling Classes, Structures and Unions](https://docs.microsoft.com/en-us/dotnet/framework/interop/marshaling-classes-structures-and-unions) documents.
That's why the explanation delivered in this pull request acts only as a brief summary or just a introduction to this.

There is also the *See also* section added, so the developer/reader knows where to search for more examples and knowledge.

Please check the commit message for more details.

### CODE REVIEW

File: docs/framework/interop/marshaling-data-with-platform-invoke.md [hunk 1]
```diff
@@ -59,3 +59,23 @@ The following code defines the library functions provided by Pinvoke.dll. Many s
 [!code-cpp[PInvokeLib#1](../../../samples/snippets/cpp/VS_Snippets_CLR/pinvokelib/cpp/pinvokelib.cpp#1)]
 
 [!code-cpp[PInvokeLib#2](../../../samples/snippets/cpp/VS_Snippets_CLR/pinvokelib/cpp/pinvokelib.h#2)]
+
+To call this in a managed code (on the C# side) you should first implement the managed prototypes for all functions and methods which are to be invoked on the C# side.
```
Comment (gewarren) [inline docs/framework/interop/marshaling-data-with-platform-invoke.md:63]:
```suggestion
To call the library functions from managed code, first implement the managed prototypes for each function you want to invoke.
```

File: docs/framework/interop/marshaling-data-with-platform-invoke.md [hunk 2]
```diff
@@ -59,3 +59,23 @@ The following code defines the library functions provided by Pinvoke.dll. Many s
...
+If an unmanaged code uses a class or struct (or any custom type) this type has to be declared also in a managed code.
```
Comment (gewarren) [inline docs/framework/interop/marshaling-data-with-platform-invoke.md:64]:
```suggestion
If the unmanaged code uses any custom types, you must also declare those types in your managed code.  
```

File: docs/framework/interop/marshaling-data-with-platform-invoke.md [hunk 3]
```diff
@@ -59,3 +59,23 @@ The following code defines the library functions provided by Pinvoke.dll. Many s
 [!code-cpp[PInvokeLib#1](../../../samples/snippets/cpp/VS_Snippets_CLR/pinvokelib/cpp/pinvokelib.cpp#1)]
 
 [!code-cpp[PInvokeLib#2](../../../samples/snippets/cpp/VS_Snippets_CLR/pinvokelib/cpp/pinvokelib.h#2)]
+
+To call this in a managed code (on the C# side) you should first implement the managed prototypes for all functions and methods which are to be invoked on the C# side.
+If an unmanaged code uses a class or struct (or any custom type) this type has to be declared also in a managed code.  
+Please see the <xref:System.Runtime.InteropServices.DllImportAttribute> which the prototype should be marked with.
```
Comment (gewarren) [inline docs/framework/interop/marshaling-data-with-platform-invoke.md:65]:
```suggestion
Decorate the prototype with the <xref:System.Runtime.InteropServices.DllImportAttribute> attribute.
```

File: docs/framework/interop/marshaling-data-with-platform-invoke.md [hunk 4]
```diff
@@ -59,3 +59,23 @@ The following code defines the library functions provided by Pinvoke.dll. Many s
...
+
+The example of such a prototype would look like:
```
Comment (gewarren) [inline docs/framework/interop/marshaling-data-with-platform-invoke.md:67]:
```suggestion
The following code shows an example prototype:
```

File: docs/framework/interop/marshaling-data-with-platform-invoke.md [hunk 5]
```diff
@@ -59,3 +59,23 @@ The following code defines the library functions provided by Pinvoke.dll. Many s
...
+
+```cs
+// Declares a managed prototype for TestStructInStruct declared and defined in unmanaged library
+[DllImport("..\\LIB\\PinvokeLib.dll", CallingConvention = CallingConvention.Cdecl)]
+internal static extern int TestStructInStruct(ref MyPerson2 person2);
+```
+
+For more examples and comprehensive explanation please see other documents in this section related to marshaling.
```
Comment (gewarren) [inline docs/framework/interop/marshaling-data-with-platform-invoke.md:75]:
```suggestion
```

File: docs/framework/interop/marshaling-data-with-platform-invoke.md [hunk 6]
```diff
@@ -59,3 +59,23 @@ The following code defines the library functions provided by Pinvoke.dll. Many s
...
+
```
Comment (gewarren) [inline docs/framework/interop/marshaling-data-with-platform-invoke.md:76]:
```suggestion
```

File: docs/framework/interop/marshaling-data-with-platform-invoke.md [hunk 7]
```diff
@@ -59,3 +59,23 @@ The following code defines the library functions provided by Pinvoke.dll. Many s
...
+## See also
+
```
Comment (gewarren) [inline docs/framework/interop/marshaling-data-with-platform-invoke.md:78]:
```suggestion

For more information and examples, see the following articles:

```

File: docs/framework/interop/marshaling-data-with-platform-invoke.md [hunk 8]
```diff
@@ -59,3 +59,23 @@ The following code defines the library functions provided by Pinvoke.dll. Many s
...
 [!code-cpp[PInvokeLib#2](../../../samples/snippets/cpp/VS_Snippets_CLR/pinvokelib/cpp/pinvokelib.h#2)]
+
+To call this in a managed code (on the C# side) you should first implement the managed prototypes for all functions and methods which are to be invoked on the C# side.
+If an unmanaged code uses a class or struct (or any custom type) this type has to be declared also in a managed code.  
+Please see the <xref:System.Runtime.InteropServices.DllImportAttribute> which the prototype should be marked with.
+
+The example of such a prototype would look like:
+
+```cs
+// Declares a managed prototype for TestStructInStruct declared and defined in unmanaged library
```
Comment (gewarren) [inline docs/framework/interop/marshaling-data-with-platform-invoke.md:70]:
```suggestion
// Managed prototype for TestStructInStruct, which is declared and defined in an unmanaged library.
```

**Review by gewarren — COMMENTED** [review]
Thank you for adding this @BartoszKlonowski !

File: docs/framework/interop/marshaling-data-with-platform-invoke.md [hunk 9]
```diff
@@ -59,3 +59,21 @@ The following code defines the library functions provided by Pinvoke.dll. Many s
...
 
 [!code-cpp[PInvokeLib#2](../../../samples/snippets/cpp/VS_Snippets_CLR/pinvokelib/cpp/pinvokelib.h#2)]
+
+To call the library functions from managed code, first implement the managed prototypes for each function you want to invoke.
+If the unmanaged code uses any any custom types, you must also declare those types in your managed code.  
+Decorate the prototype with the <xref:System.Runtime.InteropServices.DllImportAttribute> attribute.
+
+The following code shows an example prototype:
+
+```cs
```
Comment (Youssef1313) [inline docs/framework/interop/marshaling-data-with-platform-invoke.md:69]:
~~~suggestion
```csharp
~~~

**Review by gewarren — APPROVED** [review]

**Pull request merged**